calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -305,3 +306,118 @@ def test_validate_arguments(self, m_sys_exit):
+        # Assert that method exits on bad input
+        for case in cases:
+            (ip, version, cls, is_valid, case_num) = case
+            print "Testing case %s" % case_num
+            check_ip_version(ip, version, cls)
+
+            if is_valid:
+                assert not m_sys_exit.called
+            else:
+                assert m_sys_exit.called
+            m_sys_exit.reset_mock()
+
+    def test_validate_cidr(self):
+        """
+        Test validate_cidr function in calico_ctl utils
+        """
+        # Set up arguments
+        cases = (
+            ('127.a.0.1', False, 1),
+            ('aa:bb::zz', False, 2),
+            ('1.2.3.4', True, 3),
+            ('1.2.3.0/24', True, 4),
+            ('aa:bb::ff', True, 5)
+        )
+
+        # Call method under test for each test case
+        # Assert that method returns expected output
+        for case in cases:
+            (cidr, is_valid, case_num) = case
+            print "Testing case %s" % case_num
+
+            test_rtn = validate_cidr(cidr)
+
+            # Assert
+            self.assertEqual(is_valid, test_rtn)
+
+    def test_validate_ip(self):
+        """
+        Test validate_ip function in calico_ctl utils
+        """
+        # Set up arguments
+        cases = (
+            ('1.2.3.4', 'v4', True, 1),
+            ('1.2.3.4', 'v6', False, 2),
+            ('1.2.3.4', 'v4', True, 3),
+            ('1.2.3.0/24', 'v4', False, 4),
+            ('aa:bb::ff', 'v4', False, 6),
+            ('aa:bb::ff', 'v6', True, 7)
+        )
+
+        # Call method under test for each test case
+        # Assert that method returns expected output
+        for case in cases:
+            (ip, version, is_valid, case_num) = case
+            print "Testing case %s" % case_num
+
+            test_rtn = validate_ip(ip, version)
+
+            # Assert
+            self.assertEqual(is_valid, test_rtn)
+
+    def test_validate_characters(self):
+        """
+        Test validate_characters function in calico_ctl utils
+        """
+        # Set up arguments
+        cases = (
+            ('abcdefghijklmnopqrstuvwxyz', True, 1),
+            ('0123456789', True, 2),
+            ('profile_1', True, 3),
+            ('profile-1', True, 4),
+            ('profile 1', False, 5),
+            ('profile.1', True, 6),
+            ('!', False, 7),
+            ('@', False, 8),
+            ('#', False, 9),
+            ('$', False, 10),
+            ('%', False, 11),
+            ('^', False, 12),
+            ('&', False, 13),
+            ('*', False, 14),
+            ('()', False, 15)
+        )
+
+        # Call method under test for each test case
+        # Assert that method returns expected output
+        for case in cases:
+            (input_string, is_valid, case_num) = case
+            print "Testing case %s" % case_num
+
+            test_rtn = validate_characters(input_string)
+
+            # Assert
+            self.assertEqual(is_valid, test_rtn)
```

Comment:
I'd probably rename the `is_valid` var name to be more clear that it represents what you expect as output from the function.